pkg/node/manager/manager.go

```diff
@@ -401,9 +459,35 @@ func isWindowsNode(node *v1.Node) bool {
 }
 
 // canAttachTrunk returns true if the node has capability to attach a Trunk ENI
-func canAttachTrunk(node *v1.Node) bool {
+func (m *manager) canAttachTrunk(node *v1.Node) bool {
 	_, ok := node.Labels[config.HasTrunkAttachedLabel]
-	return ok
+	return ok || m.trunkEnabledInCNINode(node)
+}
+
+func (m *manager) trunkEnabledInCNINode(node *v1.Node) bool {
+	if cniNode, err := m.wrapper.K8sAPI.GetCNINode(
+		types.NamespacedName{Name: node.Name, Namespace: config.KubeDefaultNamespace},
+	); err == nil {
```

Comment:
This ignores any error cases that might be inside of GetCNINode. We should bubble those error cases to the reconcile loop, instead of assuming the trunk is not enabled.

Reply:
added error to return for bubbling up along the chain.